### PR TITLE
Fully rename to hot restart

### PR DESCRIPTION
**Pull request by mit-mit**
Follow-up change to https://github.com/flutter/flutter/commit/f48dec570c899d8e222f5b3b7cbebd7d19fe6843

### CODE REVIEW

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 1]
```diff
@@ -717,9 +717,9 @@ class HotRunner extends ResidentRunner {
     const String bold = '\u001B[0;1m';
     const String reset = '\u001B[0m';
     printStatus(
-      '$fire  To hot reload your app on the fly, press "r". To restart the app entirely, press "R".',
+      '$fire  To hot reload your app on the fly, press "r". To hot restart the app, press "R".',
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/run_hot.dart]: Given how close the names are, is it worth trying to differentiate what the two actions do here? 

`To hot reload changes into your app, press "r". To hot restart the app, press "R".`

**Review by devoncarew — APPROVED** [review]
lgtm

one comment to think about how to clarify the two actions to users in the help text, but lgtm w/o a change

File: packages/flutter_tools/lib/src/run_hot.dart (hunk 1 above)
Comment (mit-mit) [reply packages/flutter_tools/lib/src/run_hot.dart]: How about this?

`  To hot reload changes while running, press "r". To hot restart and rebuild state, press "R".`

Comment (devoncarew) [reply packages/flutter_tools/lib/src/run_hot.dart]: `To hot reload changes while running, press "r". To hot restart (and rebuild state), press "R".`?

**Pull request merged**